packages/preact/src/index.ts

```diff
@@ -388,10 +388,22 @@ export function useComputed<T>(compute: () => T) {
 	return useMemo(() => computed<T>(() => $compute.current()), []);
 }
 
+const HAS_RAF = typeof requestAnimationFrame == "function";
+function afterNextFrame(callback: () => void) {
+	let raf: number | undefined;
+
+	const done = () => {
+		clearTimeout(timeout);
+		if (HAS_RAF && raf != null) cancelAnimationFrame(raf);
+		setTimeout(callback);
+	};
+
+	const timeout = setTimeout(done, 100);
+	if (HAS_RAF) raf = requestAnimationFrame(done);
+}
```

Comment:
nit: in the case where we fall back to the timeout, we don't need to run the nested `setTimeout(0)` to wait until after the frame - there's no frame.

other nit: `requestAnimationFrame()` never returns `0`, so we can just use `if (raf)` to guard `cancelAnimationFrame` (it represents `HAS_RAF && raf != null`).

```suggestion
const HAS_RAF = typeof requestAnimationFrame == "function";
function afterNextFrame(callback: () => void) {
	let raf: number | undefined;

	const done = () => {
		clearTimeout(timeout);
		if (raf) cancelAnimationFrame(raf);
		setTimeout(callback);
	};

	const timeout = setTimeout(callback, 100);
	if (HAS_RAF) raf = requestAnimationFrame(done);
}
```

Reply:
We still need to clear the `animation` frame if there is one hence why I am calling `done` rather than `callback` but yes, `done` should not invoke another callback

```suggestion
const HAS_RAF = typeof requestAnimationFrame == "function";
function afterNextFrame(callback: () => void) {
	let raf: number | undefined;

	const done = () => {
		clearTimeout(timeout);
		if (raf != null) cancelAnimationFrame(raf);
		callback();
	};

	const timeout = setTimeout(done, 100);
	if (HAS_RAF) raf = requestAnimationFrame(done);
}
```